prime-router/src/main/kotlin/history/azure/ReportFileFunction.kt

```diff
@@ -0,0 +1,82 @@
+package gov.cdc.prime.router.history.azure
+
+import gov.cdc.prime.router.azure.WorkflowEngine
+import org.apache.logging.log4j.kotlin.Logging
+import java.time.OffsetDateTime
+import java.time.format.DateTimeParseException
+import java.util.UUID
+
+/**
+ * Submissions API
+ * Returns a list of Actions from `public.action`.
+ */
+abstract class ReportFileFunction(
+    internal val workflowEngine: WorkflowEngine = WorkflowEngine(),
+) : Logging {
+    data class Parameters(
+        val sort: String,
+        val sortColumn: String,
+        val cursor: OffsetDateTime?,
+        val endCursor: OffsetDateTime?,
```

Comment:
Why is there an endCursor? Is there a requirement to only return a certain subset of data, say Jan 1 to Mar 3?  You will need to track start/end date as separate items from cursor, since you need to be able to go both directions when paging. If this comment is unclear please let me know.

Reply:
They send the ranges as "from/to". With just those should be enough for simple forward/backwards pagination. We can't get page numbers, but that can be added later if it becomes an ask.

Reply:
That is not really how a 'cursor' works - in this case where you need to support both a start and end date and then have a cursor paging between those dates you'll need a 'startDate', 'endDate', and 'cursor' parameters. Typically cursors are used for unlimited forward/backwards paging - not within a start date.

For example, if your 'end date' was May 1 and your cursor was already at May 1 would you just...not show any more data, even if there was data?  Lets talk about this.